plugins/modules/azure_rm_manageddisk.py

```diff
@@ -460,7 +461,7 @@ def exec_module(self, **kwargs):
                     vm_name_id = self.compute_client.virtual_machines.get(vm_item['resource_group'], vm_item['name'])
                     if result['managed_by_extended'] is None or vm_name_id.id not in result['managed_by_extended']:
                         changed = True
-                        self.attach(vm_item['resource_group'], vm_item['name'], result)
+                        threading.Thread(target=self.attach, args=(vm_item['resource_group'], vm_item['name'], result)).start()
```

Comment:
Since this gets sent into a new thread, the call to `self.get_managed_disk()` on the next line may run before the thread finishes attaching the disks and it might fail because the new disk is not yet attached.

Reply:
@archoversightdefault thread has changed to thread executor